public/app/features/trails/AutomaticMetricQueries/AutoQueryEngine.ts

```diff
@@ -1,202 +1,16 @@
-import { PanelBuilders, SceneQueryRunner, VizPanelBuilder } from '@grafana/scenes';
-import { PromQuery } from 'app/plugins/datasource/prometheus/types';
-import { HeatmapColorMode } from 'app/plugins/panel/heatmap/types';
-
-import { KEY_SQR_METRIC_VIZ_QUERY, trailDS, VAR_FILTERS_EXPR, VAR_GROUP_BY_EXP, VAR_METRIC_EXPR } from '../shared';
-
-export interface AutoQueryDef {
-  variant: string;
-  title: string;
-  unit: string;
-  queries: PromQuery[];
-  vizBuilder: (def: AutoQueryDef) => VizPanelBuilder<{}, {}>;
-}
-
-export interface AutoQueryInfo {
-  preview: AutoQueryDef;
-  main: AutoQueryDef;
-  variants: AutoQueryDef[];
-  breakdown: AutoQueryDef;
-}
+import { getQueryGeneratorFor } from './query-generators';
+import { AutoQueryInfo } from './types';
 
 export function getAutoQueriesForMetric(metric: string): AutoQueryInfo {
-  let unit = 'short';
-  let agg = 'avg';
-  let rate = false;
-  let title = metric;
-
-  if (metric.endsWith('seconds_sum')) {
-    unit = 's';
-    agg = 'avg';
-    rate = true;
-  } else if (metric.endsWith('seconds')) {
-    unit = 's';
-    agg = 'avg';
-    rate = false;
-  } else if (metric.endsWith('bytes')) {
-    unit = 'bytes';
-    agg = 'avg';
-    rate = false;
-  } else if (metric.endsWith('seconds_count') || metric.endsWith('seconds_total')) {
-    agg = 'sum';
-    rate = true;
-  } else if (metric.endsWith('bucket')) {
-    return getQueriesForBucketMetric(metric);
-  } else if (metric.endsWith('count') || metric.endsWith('total')) {
-    agg = 'sum';
-    rate = true;
-  }
```

Comment:
In this `pre-refactor` logic, there is a seperate case for `seconds_count/total` and  `count/total` below, but interestingly they do the same thing -- keeping the original `short` unit.

It makes me think that at one time it had separate behavior for `seconds_...`  was set to something else, then changed. `seconds_sum` has separate behavior.

Does any of this spark a memory or why you chose one way or the other, or is it too long ago?

Either way, I'll look closely at some real work examples.